presto-main/src/main/java/com/facebook/presto/sql/analyzer/MetadataExtractor.java

```diff
@@ -236,5 +249,15 @@ protected Void visitAnalyze(Analyze node, MetadataExtractorContext context)
             context.addTable(tableName);
             return super.visitAnalyze(node, context);
         }
+
+        @Override
+        protected Void visitWith(With node, MetadataExtractorContext context)
+        {
+            node.getQueries().forEach(query -> {
+                context.addCTE(query.getName());
```

Comment:
Instead of calling it add addCTE, can we come up with some type agnostic name, such as skipProcessing(identifier)